sov-modules/sov-modules-impl/src/example_election/call.rs

```diff
@@ -74,17 +74,16 @@ impl<C: sov_modules_api::Context> Election<C> {
 
                 // Check if a candidate exist.
                 let mut candidate = candidates
-                    .get(candidate_index)
-                    .ok_or(anyhow!("Candidate doesn't exist"))?
-                    .clone();
+                    .get_mut(candidate_index)
+                    .ok_or(anyhow!("Candidate doesn't exist"))?;
 
                 candidate.count = candidate
                     .count
                     .checked_add(1)
                     .ok_or(anyhow!("Vote count overflow"))?;
 
                 // We already checked that the candidate_index exists.
-                candidates[candidate_index] = candidate;
+                candidates[candidate_index] = candidate.clone();
```

Comment:
Is this needed now? Shouldnt we be able to just remove lines 85-86?

Reply:
Yes so `candidate` is a reference but the `candidates` vec keeps owned values.

Reply:
I must be missing something - but can't we mutate the value in place through the mutable reference? If `candidate` is a `&mut Candidates[idx]`, doesn't `candidate.count = ...` already change the value in `candidates`?

Reply:
Of course we can I don't know why I didn't see it! Fixed.